### DISCUSSION TITLE
How padding/margin works in Input

### DISCUSSION THREAD

**Discussion by tamis-laan** [react-native-elements/react-native-elements#3305] [Q&A]
The input object has a 10 pixel margin or padding while for example the button object does not. 

Shouldn't all elements simply have no padding/margin at all? 

Now I have to compensate in order to make my GUI look consistent.

**Comment by NickWhiu** [comment]
I fixed this for me, using a fixed container height:

```jsx
<Input containerStyle={{ height: 55 }} />
```

**Comment by wkoutre** [comment]
> Shouldn't all elements simply have no padding/margin at all?

Agreed. Wish there was a way to easily remove all default styling.

**Comment by J6ey** [comment]
There's an annoying [bottom margin](https://snack.expo.dev/p3thKcmyy) you can't get rid of for Input. Not sure why it's there.

**Comment by wkoutre** [comment]
FWIW, here's the styling I've applied at the theme-level to the `BaseInput` in my component library (which implements `Input` from `react-native-elements`). It overrides (afaik): all height, margin, and padding-related defaults for `Input`:

```
inputContainerStyle: {
  borderBottomWidth: 0,
  height: undefined,
  paddingVertical: 0,
  paddingHorizontal: 0,
},
rightIconContainerStyle: {
  paddingLeft: 16,
  paddingRight: 0,
  paddingTop: 0,
  paddingBottom: 0,
  marginVertical: 0,
  marginHorizontal: 0,
  height: undefined,
},
leftIconContainerStyle: {
  paddingRight: 16,
  paddingLeft: 0,
  paddingTop: 0,
  paddingBottom: 0,
  marginVertical: 0,
  marginHorizontal: 0,
  height: undefined,
},
inputStyle: {
  minHeight: undefined,
  paddingVertical: 0,
  paddingHorizontal: 0,
},

```

**Comment by nicholascm** [comment]
@wkoutre It looks like this margin is still there for me even with your changes. https://snack.expo.dev/@nmorrell/jealous-cereal  . It looks like `containerStyle` lets us remove the horizontal margins, but the bottom margin stays regardless. The fixed container `height` change suggested above does limit the margin but it seems it should just be controllable via containerStyle.

**Comment by wkoutre** [comment]
@nicholascm I, too, was initially fooled by this and thought just about the same as you!

I forgot to add in my previous comment that I've also passed `renderErrorMessage={false}`. Add that to your Snack and you should see see everything laid out as you'd expect.

**Reply by AlixH** [reply]
Thank you for that !

**Reply by webintellectual** [reply]
Thank you @wkoutre

**Comment by nicholascm** [comment]
@wkoutre That is very helpful! Thanks!

**Comment by jkells** [comment]
The margin on the bottom of the Input is caused by inbuilt error message component which reserves some space. You can disable it with: `renderErrorMessage={false}`

The margin on the left and right of the Input is caused by the default 10px horizontal padding on the container. You can disable it with `containerStyle={{ paddingHorizontal: 0 }}` but you might not want to.

In my case that padding was confusing me because it looked like margin instead because I'd set a background colour on the style, whilst the containerStyle had no background making the transparent container look like margin.

To fix the issue I moved my backgroundColor from the style to the containerStyle.
